### PR TITLE
feat(backend):add Live account query and fix new create account query

### LINKED ISSUE #586: Add "Live [non-deleted] Accounts" chart
See more details about the problem space with "New Accounts" chart: https://github.com/near/near-explorer/pull/567#issuecomment-788074933

### PR DESCRIPTION
**Pull request by icerove**
fix #586

### CODE REVIEW

File: backend/src/db-utils.js [hunk 1]
```diff
@@ -281,14 +281,30 @@ const queryNewAccountsCountAggregatedByDate = async () => {
...
-      GROUP BY "date"
-      ORDER BY "date"`,
+      GROUP BY date
+      ORDER BY date`,
+    ],
+    { dataSource: DS_INDEXER_BACKEND }
+  );
+};
+
+const queryDeleteAccountAggregatedByDate = async () => {
```
Comment (frol) [inline backend/src/db-utils.js:296]:
```suggestion
const queryDeletedAccountsCountAggregatedByDate = async () => {
```

**Review by frol — COMMENTED** [review]
The query looks good to me. It took around 20 seconds on mainnet to execute.

P.S. @icerove Please, provide the execution time on testnet and mainnet whenever you introduce a new query.

**Comment by frol** [comment]
@icerove given the refactoring, it is hard to follow what was added. Please, provide me with some entries that needs to be reviewed. It seems that most of the PR is just reorganizing things (btw, given you created groups, it might be better to create a new folder, and split the file into smaller files just like you separated them into groups).

File: backend/src/db-utils.js [hunk 2]
```diff
@@ -31,6 +32,18 @@ const getSyncedGenesis = async (options) => {
   );
 };
 
+const queryGenesisAccountCount = async () => {
```
Comment (icerove) [inline backend/src/db-utils.js:35]: this is query for genesis count

File: backend/src/db-utils.js [hunk 3]
```diff
@@ -277,68 +291,53 @@ const queryTeragasUsedAggregatedByDate = async () => {
...
       GROUP BY date
       ORDER BY date`,
     ],
     { dataSource: DS_INDEXER_BACKEND }
   );
 };
 
-const queryUniqueDeployedContractsAggregatedByDate = async () => {
+// accounts
+const queryNewAccountsCountAggregatedByDate = async () => {
```
Comment (icerove) [inline backend/src/db-utils.js:316]: this is fix query for new account count

File: backend/src/db-utils.js [hunk 4]
```diff
@@ -277,68 +291,53 @@ const queryTeragasUsedAggregatedByDate = async () => {
...
+      WHERE included_in_block_timestamp < ((CAST(EXTRACT(EPOCH FROM NOW()) AS bigint) / (60 * 60 * 24)) * 60 * 60 * 24 * 1000 * 1000 * 1000)
+      GROUP BY date
       ORDER BY date`,
     ],
     { dataSource: DS_INDEXER_BACKEND }
   );
 };
 
-const queryActiveContractsCountAggregatedByDate = async () => {
+const queryDeletedAccountsCountAggregatedByDate = async () => {
```
Comment (icerove) [inline backend/src/db-utils.js:332]: this is deleted account query you should seek.

File: backend/src/stats.js [hunk 5]
```diff
@@ -93,59 +106,78 @@ async function aggregateNewAccountsCountByDate() {
   }
 }
 
-async function aggregateNewContractsCountByDate() {
+async function aggregateDeletedAccountsCountByDate() {
```
Comment (icerove) [inline backend/src/stats.js:109]: this is for aggregate deleted accounts count

File: backend/src/stats.js [hunk 6]
```diff
@@ -93,59 +106,78 @@ async function aggregateNewAccountsCountByDate() {
...
     );
-    console.log("NEW_CONTRACTS_COUNT_AGGREGATED_BY_DATE updated.");
+    console.log("DELETED_ACCOUNTS_COUNT_AGGREGATED_BY_DATE updated.");
   } catch (error) {
     console.log(error);
   }
 }
 
-async function aggregateUniqueDeployedContractsCountByDate() {
+async function aggregateLiveAccountsCountByDate() {
```
Comment (icerove) [inline backend/src/stats.js:124]: this is the most important part for this PR and check carefully for this function

File: backend/src/utils.js [hunk 7]
```diff
@@ -37,6 +37,27 @@ function formatDate(date) {
   return `${year}-${month.slice(-2)}-${day.slice(-2)} 23:59:59`;
 }
 
+function generateDateArray(startDate, endDate = new Date()) {
```
Comment (icerove) [inline backend/src/utils.js:40]: this function is for generating date array

**Comment by icerove** [comment]
@frol 
give some comments for the important functions

File: backend/src/stats.js [hunk 8]
```diff
@@ -93,59 +106,78 @@ async function aggregateNewAccountsCountByDate() {
...
+        if (newAccountMap.get(changedAccountsCountByDate[i].date)) {
+          changedAccountsCountByDate[i].accountsCount =
+            changedAccountsCountByDate[i].accountsCount +
+            newAccountMap.get(changedAccountsCountByDate[i].date);
+        }
+        if (deletedAccountMap.get(changedAccountsCountByDate[i].date)) {
+          changedAccountsCountByDate[i].accountsCount =
+            changedAccountsCountByDate[i].accountsCount -
+            deletedAccountMap.get(changedAccountsCountByDate[i].date);
+        }
```
Comment (frol) [inline backend/src/stats.js:169]:
```suggestion
        const newAccountsCount = newAccountMap.get(changedAccountsCountByDate[i].date);
        if (newAccountsCount) {
          changedAccountsCountByDate[i].accountsCount += newAccountsCount;
        }
        const deletedAccountsCount = deletedAccountMap.get(changedAccountsCountByDate[i].date);
        if (deletedAccountsCount) {
          changedAccountsCountByDate[i].accountsCount -= deletedAccountsCount;
        }
```

File: backend/src/stats.js [hunk 9]
```diff
@@ -93,59 +106,78 @@ async function aggregateNewAccountsCountByDate() {
...
+      }
+      LIVE_ACCOUNTS_COUNT_AGGREGATE_BY_DATE = changedAccountsCountByDate.reduce(
```
Comment (frol) [inline backend/src/stats.js:171]: Let's have a `cumulativeSumArray` method on the backend as well (it seems you initially copied it, but then removed it). This implementation is not reader-friendly, so it is better to have a name for it.

**Review by frol — APPROVED** [review]
Overall, looks good to me. See the nits

**Review by frol — APPROVED** [review]
